.github/workflows/dep_install_and_lint.yaml

```diff
@@ -0,0 +1,24 @@
+name: Install Dependencies and Run Linter
```

Comment:
Seems duplicated from `.github/actions/dep_install_and_lint/action.yaml`, maybe we can use the action in this workflow?

Reply:
yes, this should be deleted!